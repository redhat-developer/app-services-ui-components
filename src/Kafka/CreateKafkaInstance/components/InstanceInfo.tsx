import {
  Button,
  ButtonVariant,
  Card,
  CardBody,
  CardTitle,
  DescriptionList,
  DescriptionListDescription,
  DescriptionListGroup,
  DescriptionListTerm,
  Grid,
  GridItem,
  Stack,
  StackItem,
} from "@patternfly/react-core";

import { ClockIcon } from "@patternfly/react-icons";
import type { VoidFunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import type { MarketplaceSubscription } from "../../types";
import type { BillingHelpProps } from "./BillingHelp";
import { BillingHelp } from "./BillingHelp";

export type InstanceInfoProps = {
  isTrial: boolean;
  onClickQuickStart: () => void;

  trialDurationInHours: number | undefined;
  ingress: number;
  egress: number;
  /**
   * Max storage in GiB
   */
  storage: number;
  maxPartitions: number;
  connections: number;
  /**
   * Max connections/second
   */
  connectionRate: number;
  /**
   * Message size in MiB
   */
  messageSize: number;
  streamingUnits: string | undefined;

  billing:
    | ({
        value: "prepaid" | MarketplaceSubscription;
      } & BillingHelpProps)
    | undefined;
};

export const InstanceInfo: VoidFunctionComponent<InstanceInfoProps> = ({
  isTrial,
  trialDurationInHours,
  ingress,
  egress,
  storage,
  maxPartitions,
  connections,
  connectionRate,
  messageSize,
  onClickQuickStart,
  streamingUnits,
  billing,
}) => {
  const { t } = useTranslation("create-kafka-instance");

  return (
    <Stack hasGutter data-testid={"instance-info"}>
      <StackItem>
        <Card isFlat>
          <CardTitle component="h2">{t("instance_information")}</CardTitle>
          <CardBody>
            <DescriptionList isCompact>
              <DescriptionListGroup>
                <Grid sm={6} lg={12} hasGutter>
                  {!isTrial && (
                    <GridItem data-testid={"instance-size"}>
                      <DescriptionListTerm>
                        {t("common:size")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("stream_unit_value", { size: streamingUnits })}
                      </DescriptionListDescription>
                    </GridItem>
                  )}
                  {isTrial && (
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:duration")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        <ClockIcon color="var(--pf-global--info-color--100)" />{" "}
                        {t("duration_value", {
                          value: trialDurationInHours,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                  )}
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:ingress")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("ingress_value", {
                        value: ingress,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:egress")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("egress_value", {
                        value: egress,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:storage")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("storage_value", {
                        value: storage,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:partitions")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("partitions_value", {
                        value: maxPartitions,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:client_connections")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("client_connections_value", {
                        value: connections,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:connection_rate")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("connection_rate_value", {
                        value: connectionRate,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  <GridItem>
                    <DescriptionListTerm>
                      {t("kafka:message_size")}
                    </DescriptionListTerm>
                    <DescriptionListDescription>
                      {t("message_size_value", {
                        value: messageSize,
                      })}
                    </DescriptionListDescription>
                  </GridItem>
                  {billing && (
                    <GridItem>
                      <DescriptionListTerm>
                        {t("billing.field_label")}&nbsp;
                        <BillingHelp
                          type={billing.type}
                          subscriptionOptionsHref={
                            billing.subscriptionOptionsHref
                          }
                        />
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {(() => {
                          if (billing.value === "prepaid") {
                            return t("billing.prepaid_option");
                          }
                          switch (billing.value.marketplace) {
                            case "aws":
                              return (
                                <>
                                  {t("billing.marketplace_aws")}
                                  <br />
                                  {billing.value.subscription}
                                </>
                              );
                            case "gcp":
                              return (
                                <>
                                  {t("billing.marketplace_gcp")}
                                  <br />
                                  {billing.value.subscription}
                                </>
                              );
                            case "azure":
                              return (
                                <>
                                  {t("billing.marketplace_azure")}
                                  <br />
                                  {billing.value.subscription}
                                </>
                              );
                            case "rhm":
                              return (
                                <>
                                  {t("billing.marketplace_rh")}
                                  <br />
                                  {billing.value.subscription}
                                </>
                              );
                          }
                        })()}
                      </DescriptionListDescription>
                    </GridItem>
                  )}
                </Grid>
              </DescriptionListGroup>
            </DescriptionList>
          </CardBody>
        </Card>
      </StackItem>
      <StackItem>
        <Card isFlat isCompact>
          <CardTitle component="h2">{t("quick_start_guide_title")}</CardTitle>
          <CardBody>
            <Button
              isSmall
              isInline
              variant={ButtonVariant.link}
              onClick={onClickQuickStart}
            >
              {t("quick_start_guide_message_exp")}
            </Button>
          </CardBody>
        </Card>
      </StackItem>
    </Stack>
  );
};
