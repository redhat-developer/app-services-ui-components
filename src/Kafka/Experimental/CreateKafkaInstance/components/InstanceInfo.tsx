import { VoidFunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import {
  Button,
  ButtonVariant,
  Grid,
  GridItem,
  Skeleton,
  Card,
  CardBody,
  CardTitle,
  Stack,
  StackItem,
  DescriptionList,
  DescriptionListTerm,
  DescriptionListGroup,
  DescriptionListDescription,
} from "@patternfly/react-core";

import ClockIcon from "@patternfly/react-icons/dist/js/icons/outlined-clock-icon";

export type InstanceInfoProps = {
  isLoading: boolean;
  isTrial: boolean;
  onClickQuickStart: () => void;
};

export type InstanceInfoLimitsProps = {
  trialDurationInHours: number;
  ingresEgress: number;
  /**
   * Max storage in GiB
   */
  storage: number;
  maxPartitions: number;
  connections: number;
  /**
   * Max connections/second
   */
  connectionRate: number;
  /**
   * Message size in MiB
   */
  messageSize: number;
  streamingUnits: number;
};

export const InstanceInfo: VoidFunctionComponent<
  InstanceInfoProps & InstanceInfoLimitsProps
> = ({
  isLoading,
  isTrial,
  trialDurationInHours,
  ingresEgress,
  storage,
  maxPartitions,
  connections,
  connectionRate,
  messageSize,
  onClickQuickStart,
  streamingUnits,
}) => {
  const { t } = useTranslation("create-kafka-instance");

  return (
    <Stack hasGutter data-testid={"instance-info"}>
      <StackItem>
        <Card isFlat>
          <CardTitle component="h2">{t("instance_information")}</CardTitle>
          <CardBody>
            <DescriptionList isCompact>
              <DescriptionListGroup>
                {!isLoading && (
                  <Grid sm={6} lg={12} hasGutter>
                    {!isTrial && (
                      <GridItem>
                        <DescriptionListTerm>
                          {t("common:size")}
                        </DescriptionListTerm>
                        <DescriptionListDescription>
                          {t("stream_unit_value", { size: streamingUnits })}
                        </DescriptionListDescription>
                      </GridItem>
                    )}
                    {isTrial && (
                      <GridItem>
                        <DescriptionListTerm>
                          {t("kafka:duration")}
                        </DescriptionListTerm>
                        <DescriptionListDescription>
                          <ClockIcon color="var(--pf-global--info-color--100)" />{" "}
                          {t("duration_value", {
                            value: trialDurationInHours,
                          })}
                        </DescriptionListDescription>
                      </GridItem>
                    )}
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:ingress")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("ingress_value", {
                          value: ingresEgress,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:egress")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("egress_value", {
                          value: ingresEgress,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:storage")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("storage_value", {
                          value: storage,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:partitions")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("partitions_value", {
                          value: maxPartitions,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:client_connections")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("client_connections_value", {
                          value: connections,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:connection_rate")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("connection_rate_value", {
                          value: connectionRate,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                    <GridItem>
                      <DescriptionListTerm>
                        {t("kafka:message_size")}
                      </DescriptionListTerm>
                      <DescriptionListDescription>
                        {t("message_size_value", {
                          value: messageSize,
                        })}
                      </DescriptionListDescription>
                    </GridItem>
                  </Grid>
                )}
                {isLoading && (
                  <Grid sm={6} lg={12} hasGutter>
                    <GridItem>
                      <Skeleton
                        width="75%"
                        screenreaderText="Loading contents"
                      />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="66%" />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="66%" />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="50%" />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="33%" />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="25%" />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="55%" />
                    </GridItem>
                    <GridItem>
                      <Skeleton width="35%" />
                    </GridItem>
                  </Grid>
                )}
              </DescriptionListGroup>
            </DescriptionList>
          </CardBody>
        </Card>
      </StackItem>
      <StackItem>
        <Card isFlat isCompact>
          <CardTitle component="h2">{t("quick_start_guide_title")}</CardTitle>
          <CardBody>
            <Button
              isSmall
              isInline
              variant={ButtonVariant.link}
              onClick={onClickQuickStart}
            >
              {t("quick_start_guide_message_exp")}
            </Button>
          </CardBody>
        </Card>
      </StackItem>
    </Stack>
  );
};
