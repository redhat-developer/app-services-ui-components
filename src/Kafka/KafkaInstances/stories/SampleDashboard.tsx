export default () => <div>Dashboard</div>;
