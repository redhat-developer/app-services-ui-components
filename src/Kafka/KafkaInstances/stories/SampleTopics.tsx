export default () => <div>Topics</div>;
