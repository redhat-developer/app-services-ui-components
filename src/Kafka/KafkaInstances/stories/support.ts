import { addHours } from "date-fns";
import { useEffect, useState } from "react";
import type { KafkaInstance } from "../../types";

const now = new Date().toISOString();
const future = addHours(new Date(), 19).toISOString();
const INSTANCES: KafkaInstance[] = [
  {
    id: "1",
    name: "foo",
    createdAt: now,
    updatedAt: now,
    expiryDate: undefined,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "provisioning",
    plan: "standard",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "2",
    name: "bar",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "deleting",
    plan: "developer",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "3",
    name: "baz",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "azure",
    region: "region",
    status: "ready",
    plan: "standard",
    size: "2",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
];
export const useInstances = () => {
  const [instances, setInstances] = useState<KafkaInstance[]>();

  useEffect(() => {
    const timer = setTimeout(() => {
      setInstances(INSTANCES);
    }, 500);
    return () => clearTimeout(timer);
  }, []);
  return {
    instances,
    loading: instances === undefined,
  };
};
export const useInstance = (id: string) => {
  const { instances, loading } = useInstances();
  const instance = instances?.find((i) => i.id === id);

  return {
    instance,
    loading: loading || instance === undefined,
  };
};
