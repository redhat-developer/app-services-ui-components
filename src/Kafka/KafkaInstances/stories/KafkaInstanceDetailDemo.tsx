import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
