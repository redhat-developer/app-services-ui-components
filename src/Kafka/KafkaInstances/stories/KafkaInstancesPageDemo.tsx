import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
