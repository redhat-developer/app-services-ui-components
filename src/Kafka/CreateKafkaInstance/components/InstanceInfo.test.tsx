import { render, waitForI18n } from "../../../test-utils";
import { composeStories } from "@storybook/testing-react";
import * as stories from "./InstanceInfo.stories";

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName!,
  Story,
]);

// Batch snapshot testing
test.each(testCases)("Renders %s story", async (_storyName, Story) => {
  const tree = await render(<Story />);
  await waitForI18n(tree);
  expect(tree.baseElement).toMatchSnapshot();
});
