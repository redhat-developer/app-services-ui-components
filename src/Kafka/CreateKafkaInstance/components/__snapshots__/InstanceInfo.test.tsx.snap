// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceInfo Renders QuotaInstance story 1`] = `
<body>
  <div>
    <div
      class="pf-m-flex-1 mas--CreateKafkaInstance__sidebar"
      style=""
    >
      <div
        class="pf-l-stack pf-m-gutter"
        data-testid="instance-info"
      >
        <div
          class="pf-l-stack__item"
        >
          <article
            class="pf-c-card pf-m-flat"
            data-ouia-component-id="OUIA-Generated-Card-1"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <h2
              class="pf-c-card__title"
            >
              Details
            </h2>
            <div
              class="pf-c-card__body"
            >
              <dl
                class="pf-c-description-list pf-m-compact"
              >
                <div
                  class="pf-c-description-list__group"
                >
                  <div
                    class="pf-l-grid pf-m-all-6-col-on-sm pf-m-all-12-col-on-lg pf-m-gutter"
                  >
                    <div
                      class="pf-l-grid__item"
                      data-testid="instance-size"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Size
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          1 streaming units
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Ingress
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 MiB/second
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Egress
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 MiB/second
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Storage
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 GiB
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Partitions
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Client connections
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Connection rate
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 connections/second
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Message size
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 MiB
                        </div>
                      </dd>
                    </div>
                  </div>
                </div>
              </dl>
            </div>
          </article>
        </div>
        <div
          class="pf-l-stack__item"
        >
          <article
            class="pf-c-card pf-m-compact pf-m-flat"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <h2
              class="pf-c-card__title"
            >
              Need help getting started?
            </h2>
            <div
              class="pf-c-card__body"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline pf-m-small"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Follow our quick start guide.
              </button>
            </div>
          </article>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`InstanceInfo Renders TrialInstance story 1`] = `
<body>
  <div>
    <div
      class="pf-m-flex-1 mas--CreateKafkaInstance__sidebar"
      style=""
    >
      <div
        class="pf-l-stack pf-m-gutter"
        data-testid="instance-info"
      >
        <div
          class="pf-l-stack__item"
        >
          <article
            class="pf-c-card pf-m-flat"
            data-ouia-component-id="OUIA-Generated-Card-3"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <h2
              class="pf-c-card__title"
            >
              Details
            </h2>
            <div
              class="pf-c-card__body"
            >
              <dl
                class="pf-c-description-list pf-m-compact"
              >
                <div
                  class="pf-c-description-list__group"
                >
                  <div
                    class="pf-l-grid pf-m-all-6-col-on-sm pf-m-all-12-col-on-lg pf-m-gutter"
                  >
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Duration
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          <svg
                            aria-hidden="true"
                            fill="var(--pf-global--info-color--100)"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M256,8C119,8,8,119,8,256S119,504,256,504,504,393,504,256,393,8,256,8Zm92.49,313h0l-20,25a16,16,0,0,1-22.49,2.5h0l-67-49.72a40,40,0,0,1-15-31.23V112a16,16,0,0,1,16-16h32a16,16,0,0,1,16,16V256l58,42.5A16,16,0,0,1,348.49,321Z"
                            />
                          </svg>
                           
                          123 hours
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Ingress
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 MiB/second
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Egress
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 MiB/second
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Storage
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 GiB
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Partitions
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Client connections
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Connection rate
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 connections/second
                        </div>
                      </dd>
                    </div>
                    <div
                      class="pf-l-grid__item"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          Message size
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          up to 123 MiB
                        </div>
                      </dd>
                    </div>
                  </div>
                </div>
              </dl>
            </div>
          </article>
        </div>
        <div
          class="pf-l-stack__item"
        >
          <article
            class="pf-c-card pf-m-compact pf-m-flat"
            data-ouia-component-id="OUIA-Generated-Card-4"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <h2
              class="pf-c-card__title"
            >
              Need help getting started?
            </h2>
            <div
              class="pf-c-card__body"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-link pf-m-inline pf-m-small"
                data-ouia-component-id="OUIA-Generated-Button-link-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Follow our quick start guide.
              </button>
            </div>
          </article>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
